auth/simple_token.go

```diff
@@ -100,7 +104,15 @@ func (tm *simpleTokenTTLKeeper) run() {
 	}
 }
 
-func (as *authStore) GenSimpleToken() (string, error) {
+type tokenSimple struct {
+	as *authStore
```

Comment:
I don't think this needs a dependency on `authStore` either. `isValidSimpleToken` can be folded into `tokenSimple` by having `func NewSimpleTokenProvider(indexWaiter func(int64) <-chan struct{}) *tokenSimple`